scan.py

```diff
@@ -46,13 +46,33 @@
 
 
 def event_object(event, s3_resource=None):
-    bucket = json.loads(event["Records"][0]["Sns"]["Message"])["Records"][0]["s3"]["bucket"]["name"]
-    key = unquote_plus(json.loads(event["Records"][0]["Sns"]["Message"])["Records"][0]["s3"]["object"]["key"])
-
-    if (not bucket) or (not key):
+    bucket = None
+    key = None
+
+    # check that the event is properly formatted
+    if "Records" in event and len(event["Records"]) > 0:
+        # handle sns messages
+        if "EventSource" in event["Records"][0] and event["Records"][0]["EventSource"] == "aws:sns":
+            print("Handling SNS message")
+            
+            payload = json.loads(event["Records"][0]["Sns"]["Message"])
+            bucket = payload["Records"][0]["s3"]["bucket"]["name"]
+            key = unquote_plus(payload["Records"][0]["s3"]["object"]["key"])
```

Comment:
maybe just return inside of the if or elif that way you don't need to check for bucket/key None - just raise if you make it past the if/elif in a final else statement?